### PR TITLE
Assets Precompile in Rails Guide is NOT showing the steps for Sprockets 4

### LINKED ISSUE #46184: Assets Precompile in Rails Guide is NOT showing the steps for Sprockets 4
### Steps to reproduce

The guide at https://guides.rubyonrails.org/asset_pipeline.html#precompiling-assets documents Sprockets 3 setup. Whereas Sprockets 4 uses `manifest.js` to configure precompile list as per https://github.com/rails/sprockets/blob/main/UPGRADING.md#manifestjs.

### Expected behavior

Perhaps add a section, or link to Sprockets 4 setup, and have a heading to say https://guides.rubyonrails.org/asset_pipeline.html#precompiling-assets is referring to Sprockets 3.

### Actual behavior

No mention of Sprocket version in Guide.

### PR DESCRIPTION
**Pull request by matiassalles99**
### Motivation / Background

Tackling #46184. As khoan mentions, https://guides.rubyonrails.org/asset_pipeline.html#precompiling-assets documents Sprockets 3 setup whereas Sprockets 4 uses manifest.js to configure precompile list as per https://github.com/rails/sprockets/blob/main/UPGRADING.md#manifestjs.

Fixes #46184

### Detail

This Pull Request changes this file https://github.com/rails/rails/blob/main/guides/source/asset_pipeline.md and differentiates sprockets version 3 and 4 setup

### CODE REVIEW

**Comment by ghiculescu** [comment]
Thanks! Could you change your PR body to include "Fixes #46184", then this will auto-close the issue when it's merged.

File: guides/source/asset_pipeline.md
Comment (ghiculescu) [inline guides/source/asset_pipeline.md:671]:
```suggestion
- **Sprockets 3:** Add files to `precompile` array in `config/initializers/assets.rb`:

  ```ruby
  Rails.application.config.assets.precompile += %w( admin.js admin.css )
  ```

- **Sprockets 4:** Add files or directories inside `./app/assets/config/manifest.js`

  ```js
  //= link_tree ../images
  //= link_directory ../javascripts .js
  //= link_directory ../stylesheets .css
  ... (other files/directories)
  ```
```

**Comment by matiassalles99** [comment]
Sure!

File: guides/source/asset_pipeline.md
Comment (matiassalles99) [reply guides/source/asset_pipeline.md]: Just pushed the changes, also noticed that .gitignore does not have .DS_Store included, is there a reason for it?

**Comment by ghiculescu** [comment]
Looks good. Please squash your commits and then we'll have to wait for someone from the core team to review 👍

File: guides/source/asset_pipeline.md
Comment (p8) [inline guides/source/asset_pipeline.md:661]:
```suggestion
- **Sprockets 3:** Add files to the `precompile` array in `config/initializers/assets.rb`:
```

File: guides/source/asset_pipeline.md
Comment (p8) [inline guides/source/asset_pipeline.md:674]: I think it is clearer if we use the same admin assets in the example:
```suggestion
  ```js
  //= link_tree ../images
  //= link admin.js
  //= link admin.css
  //= link_directory ../javascripts .js
  //= link_directory ../stylesheets .css
  ```
```

File: guides/source/asset_pipeline.md
Comment (p8) [inline guides/source/asset_pipeline.md:679]: Instead of using `here` we can already describe what it links to:
```suggestion
NOTE. More information on the differences between Sprockets 3 and 4 please can be found in the [Sprockets Upgrading Guide](https://github.com/rails/sprockets/blob/main/UPGRADING.md#manifestjs)
```

**Review by p8 — CHANGES_REQUESTED** [review]

**Comment by matiassalles99** [comment]
Thanks for the suggestions! should I squash all the commits?

**Pull request merged**
